README.md

```diff
@@ -65,7 +65,7 @@ On Windows you will need [Microsoft C++ Build Tools](https://visualstudio.micros
 ```bash
 pip install --upgrade pip
 pip install --upgrade setuptools
-pip install recommenders[examples]
+pip install --no-cache --no-binary scikit-surprise recommenders[examples]
```

Comment:
Yes, we need this because we want pip to build Surprise from the source (I changed `setup.py` in an earlier PR to link to the source on GitHub). Otherwise we get the conflict between Surprise and Numpy 1.19 (array size mismatch).
Maybe there is a way to do this inside `setup.py`, any ideas?

Reply:
@NicolasHug hey Nicolas, we are upgrading to TF 2 in recommenders, we are having a conflict between surprise and TF 2. 

What @anargyri is doing at the moment is adding the [tarball](https://github.com/microsoft/recommenders/blob/andreas/tf/setup.py#L53) and also we have to install `--no-cache --no-binary scikit-surprise` before recommenders. 

Do you know what is the best way to install surprise with TF2 and add it in the `extra_requires` like we had [before](https://github.com/microsoft/recommenders/blob/main/setup.py#L46)?

Reply:
@miguelgfierro See [this issue and discussion](https://github.com/microsoft/recommenders/issues/1511#issuecomment-929198497)

Reply:
Hi all, sorry for the troubles - any chance you could just rely on conda-forge to install surprise binaries directly instead of building from source?

Reply:
Hi @NicolasHug no worries about this. I think what you suggest could work within a conda environment, however we have an installation process using pip for recommenders, which can work in both conda and virtualenv. 
To summarize the issue, numpy 1.19 has an inconsistency with the binaries from surprise 1.1.1 (numpy 1.20 is fine). The inconsistency is related to the .pyx files, so building from source solves the issue.

Reply:
Hi @NicolasHug, would you mind creating a micro release of Surprise by pip installing numpy 1.19 for the python setup.py ? 
I think this would be a nice solution to our issue but also to anyone who would like to pip install scikit-surprise but needs numpy 1.19 for whatever reason (e.g. because of TensorFlow).

Reply:
I'll try to look into this this WE

Reply:
Thanks, we'd really appreciate it 🙏

Reply:
Thanks for your support @NicolasHug!

Reply:
I had time to look into this a little. I'm a bit surprised that we're getting this "size changed" issue. According to https://github.com/numpy/numpy/issues/18709, this issue happens when one is trying to run code that was compiled against a more recent numpy version than what is currently installed.

But since I released surprise 1.1.1 in July 2020, the latest possible version I could have possibly used is numpy 1.19 (which was released one month earlier).

So I'm not quite sure why you're getting this error with 1.19 TBH. Is it possible that you're installing / building surprise with e.g. numpy 1.20/1.21 and then downgrade to numpy 1.19? If that's the case, then the error is expected.

On my end, I'm only able to reproduce the issue if I build on (e.g.) 1.20 and then try to import surprise with 1.19. If I build with 1.17, I can import surprise with all versions all the way up to 1.21.

----

Anyway, it's possible that I'm missing something. Since 1.17 seems to work fine upwards (and since it's older and thus less constraining than 1.19 in terms of dependency satisfaction), I uploaded a new release where Cython files were compiled against 1.17. I only updated it to the pypi-test channel for now. Before I push it for good on the official PyPI channel, could you please check that it actually works for you guys?  You can install from pypi-test with e.g.

`pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple scikit-surprise`

Note: when trying that locally you might need a `--no-cache-dir` flag to make sure stuff gets re-compiled when you switch numpy versions.

Reply:
Thanks @NicolasHug, I still get the same behavior with the new release, however. As before, pip installing without the no-binary option leads to the error, using this option works fine. 
According to the issue you linked, it seems that others have been affected by this too. It looks like this behavior is not really about the version of numpy itself but has to do with the Cython compilation. So that everything works when `python setup.py` is run during the pip installation (which is the effect of the no-binary option). Gensim has observed the same. 
Maybe we should wait until there is a fix by numpy or cython. Until then, we could require users to use `--no-binary`. Thanks for trying anyway!

Reply:
Hi @NicolasHug FYI I noticed something related, which I posted [here](https://github.com/NicolasHug/Surprise/issues/397).

Reply:
@NicolasHug @miguelgfierro I did some further investigation. To summarize,

```
pip install numpy==1.19 cython
pip install scikit-surprise
```
installs but then importing surprise raises ValueError.

```
pip install numpy==1.19 cython
pip install scikit-surprise@https://github.com/NicolasHug/Surprise/archive/refs/tags/v1.1.1.tar.gz
```
works correctly.
So there is a workaround for people who want to use numpy <=1.19 along with surprise. 
In our case, it still doesn't work via our `setup.py` (unless you pre-install cython), but this must be some pip / setuptools bug.